diff -Nru tor-0.2.3.25_ORIG/GENERA_LIBTOR_BOT.sh tor-0.2.3.25/GENERA_LIBTOR_BOT.sh
--- tor-0.2.3.25_ORIG/GENERA_LIBTOR_BOT.sh	1970-01-01 01:00:00.000000000 +0100
+++ tor-0.2.3.25/GENERA_LIBTOR_BOT.sh	2013-01-27 21:53:59.016059516 +0100
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+echo "En que puerto quieres que escuche tor? "
+read PUERTO_TOR
+if [ ! -z $PUERTO_TOR ]; then
+	sed -r "s/BOT_SOCKSPORT [0-9]+/BOT_SOCKSPORT $PUERTO_TOR/" src/or/config.c > src/or/config_new.c
+	mv src/or/config_new.c src/or/config.c
+fi
+make distclean
+export CFLAGS="-s -Wno-unused-value"
+./configure
+make -j 2
+rm libtor.a
+ar crv libtor.a $(find -type f -name *.o)
+echo "*{{ libtor.a CREADA }}*"
+echo "!! Recuerda que libtor.a requiere estas librerías: -lz -lm -levent -lssl -lcrypto -lpthread -ldl !!"
diff -Nru tor-0.2.3.25_ORIG/src/common/torlog.h tor-0.2.3.25/src/common/torlog.h
--- tor-0.2.3.25_ORIG/src/common/torlog.h	2012-11-19 22:24:38.000000000 +0100
+++ tor-0.2.3.25/src/common/torlog.h	2013-01-27 21:52:24.464008022 +0100
@@ -9,6 +9,8 @@
  *
  * \brief Headers for log.c
  **/
+/* !! CONTEXTO BOT!! */
+#define __BOT
 
 #ifndef _TOR_LOG_H
 
@@ -151,6 +153,20 @@
 
 void tor_log(int severity, log_domain_mask_t domain, const char *format, ...)
   CHECK_PRINTF(3,4);
+
+#ifdef __BOT
+
+/* Inhibimos todo tipo de log */
+#define log_fn 
+#define log_debug 
+#define log_info 
+#define log_notice 
+#define log_warn 
+#define log_err 
+#define log
+
+#else /* Si no está definido __BOT dejamos todo como estaba */
+
 #define log tor_log /* hack it so we don't conflict with log() as much */
 
 #if defined(__GNUC__) || defined(RUNNING_DOXYGEN)
@@ -211,6 +227,8 @@
 
 #endif /* !GNUC */
 
+#endif /* __BOT */
+
 # define _TOR_LOG_H
 #endif
 
diff -Nru tor-0.2.3.25_ORIG/src/or/config.c tor-0.2.3.25/src/or/config.c
--- tor-0.2.3.25_ORIG/src/or/config.c	2012-11-19 22:24:38.000000000 +0100
+++ tor-0.2.3.25/src/or/config.c	2013-01-27 21:52:24.468008026 +0100
@@ -4640,58 +4640,22 @@
     argv = backup_argv;
     argc = backup_argc;
   }
-  if (argc > 1 && (!strcmp(argv[1], "-h") || !strcmp(argv[1],"--help"))) {
-    print_usage();
-    exit(0);
-  }
-  if (argc > 1 && !strcmp(argv[1], "--list-torrc-options")) {
-    /* For documenting validating whether we've documented everything. */
-    list_torrc_options();
-    exit(0);
-  }
-
-  if (argc > 1 && (!strcmp(argv[1],"--version"))) {
-    printf("Tor version %s.\n",get_version());
-    exit(0);
-  }
-  if (argc > 1 && (!strcmp(argv[1],"--digests"))) {
-    printf("Tor version %s.\n",get_version());
-    printf("%s", libor_get_digests());
-    printf("%s", tor_get_digests());
-    exit(0);
-  }
-
-  /* Go through command-line variables */
-  if (!global_cmdline_options) {
-    /* Or we could redo the list every time we pass this place.
-     * It does not really matter */
-    if (config_get_commandlines(argc, argv, &global_cmdline_options) < 0) {
-      goto err;
-    }
-  }
 
   command = CMD_RUN_TOR;
-  for (i = 1; i < argc; ++i) {
-    if (!strcmp(argv[i],"--list-fingerprint")) {
-      command = CMD_LIST_FINGERPRINT;
-    } else if (!strcmp(argv[i],"--hash-password")) {
-      command = CMD_HASH_PASSWORD;
-      command_arg = tor_strdup( (i < argc-1) ? argv[i+1] : "");
-      ++i;
-    } else if (!strcmp(argv[i],"--verify-config")) {
-      command = CMD_VERIFY_CONFIG;
-    }
-  }
 
-  if (command == CMD_HASH_PASSWORD) {
-    cf = tor_strdup("");
+  #define BOT_SOCKSPORT 9999
+  #define BOT_TORCONF(buf, puerto) { \
+                                     buf = (char *)malloc(128); \
+                                     sprintf(buf, "SocksPort %d\nSocksListenAddress 127.0.0.1\nAvoidDiskWrites 1\n", puerto); \
+                                   }
+  /* Adesso possiamo passare la porta da riga di comando */
+  if (argc == 1) {
+      BOT_TORCONF(cf, atoi(argv[0]));
   } else {
-    cf_defaults = load_torrc_from_disk(argc, argv, 1);
-    cf = load_torrc_from_disk(argc, argv, 0);
-    if (!cf)
-      goto err;
+      BOT_TORCONF(cf, BOT_SOCKSPORT);
   }
 
+  /* Ahora tenemos en cf una configuración como si hubiese leido de un fichero torrc ejemplo ;) */
   retval = options_init_from_string(cf_defaults, cf, command, command_arg,
                                     &errmsg);
 
diff -Nru tor-0.2.3.25_ORIG/src/or/main.c tor-0.2.3.25/src/or/main.c
--- tor-0.2.3.25_ORIG/src/or/main.c	2012-11-19 22:24:54.000000000 +0100
+++ tor-0.2.3.25/src/or/main.c	2013-01-27 21:52:24.472008030 +0100
@@ -2281,12 +2281,8 @@
 
   /* We search for the "quiet" option first, since it decides whether we
    * will log anything at all to the command line. */
-  for (i=1;i<argc;++i) {
-    if (!strcmp(argv[i], "--hush"))
-      quiet = 1;
-    if (!strcmp(argv[i], "--quiet"))
-      quiet = 2;
-  }
+   
+  quiet = 2; /* Siempre en quieto!! */
  /* give it somewhere to log to initially */
   switch (quiet) {
     case 2:
diff -Nru tor-0.2.3.25_ORIG/src/or/tor_main.c tor-0.2.3.25/src/or/tor_main.c
--- tor-0.2.3.25_ORIG/src/or/tor_main.c	2012-10-14 00:34:46.000000000 +0200
+++ tor-0.2.3.25/src/or/tor_main.c	2013-01-27 21:52:24.475008033 +0100
@@ -21,12 +21,12 @@
 
 int tor_main(int argc, char *argv[]);
 
-/** We keep main() in a separate file so that our unit tests can use
- * functions from main.c)
- */
-int
-main(int argc, char *argv[])
+/* Función que lanza el tor por librería */
+void lanza_tor(char *port)
 {
-  return tor_main(argc, argv);
+   if (port)
+       tor_main(1, &port);
+   else
+       tor_main(0, 0);
 }
 
